internal/controllers/reconciliation/controller.go

```diff
@@ -156,19 +157,20 @@ func (c *Controller) Reconcile(ctx context.Context, req *reconstitution.Request)
 	}
 
 	// Store the results
+	if modified {
+		return ctrl.Result{Requeue: true}, nil
+	}
+	deleted := current == nil || current.GetDeletionTimestamp() != nil
 	c.resourceClient.PatchStatusAsync(ctx, &req.Manifest, func(rs *apiv1.ResourceState) *apiv1.ResourceState {
-		if rs.Deleted == resource.Deleted() && rs.Reconciled && rs.Ready != nil && *rs.Ready == *ready {
+		if rs != nil && rs.Deleted == deleted && rs.Reconciled && ptr.Deref(rs.Ready, metav1.Time{}) == ptr.Deref(ready, metav1.Time{}) {
```

Comment:
Deletion state is now the actual deletion state of the resource, not a reflection of if we've successfully reconciled our desired state. The two values diverge when resources are purposefully orphaned, or likely in other cases (at least other bugs).